### PR TITLE
ignore invalid characters when getting decoding error in cmd run

### PR DESCRIPTION
**Pull request by mcalmer**
### What does this PR do?

When you use cmd.run and the command return latin-1 characters, salt just throw the UnicodeDecodeError. 
This PR write an error message and call decode() with `ignore` to get at least a bit of the output.

### Tests written?

No

### Commits signed with GPG?

No

### CODE REVIEW

**Comment by rallytime** [comment]
re-run py3

File: salt/modules/cmdmod.py [hunk 1]
```diff
@@ -589,11 +589,17 @@ def _get_stripped(cmd):
             out = proc.stdout.decode(__salt_system_encoding__)
         except AttributeError:
             out = u''
+        except UnicodeDecodeError:
+            log.error("UnicodeDecodeError while decoding output of cmd {0}".format(cmd))
+            out = proc.stdout.decode(__salt_system_encoding__, 'ignore')
```
Comment (damon-atkins) [inline salt/modules/cmdmod.py]: replace might be better than ignore. At least the user will be more aware of it. Might be worth adding the ability to override `__salt_system_encoding__`

Comment (mcalmer) [reply salt/modules/cmdmod.py]: I like replace. Changed and rebased

File: salt/modules/cmdmod.py [hunk 2]
```diff
@@ -589,11 +589,18 @@ def _get_stripped(cmd):
             out = proc.stdout.decode(__salt_system_encoding__)
         except AttributeError:
             out = u''
+        except UnicodeDecodeError:
+            log.error("UnicodeDecodeError while decoding output of cmd {0}".format(cmd))
```
Comment (damon-atkins) [inline salt/modules/cmdmod.py]: The standard is `'` not `"`

**Review by damon-atkins — CHANGES_REQUESTED** [review]

File: salt/modules/cmdmod.py (hunk 2 above)
Comment (mcalmer) [reply salt/modules/cmdmod.py]: fixed

**Comment by rallytime** [comment]
@mcalmer There's a small lint error here: https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/16781/violations/file/salt/modules/cmdmod.py/

**Review by terminalmage — APPROVED** [review]

**Pull request merged**